src/OpenTelemetry/Internal/OpenTelemetrySdkEventSource.cs

```diff
@@ -358,7 +358,7 @@ protected override void OnEventWritten(EventWrittenEventArgs e)
                     message = e.Message;
                 }
 
-                Debug.WriteLine($"{e.EventSource.Name} - EventId: [{e.EventId}], EventName: [{e.EventName}], Message: [{message}]");
+                Debug.WriteLine($"{e.EventSource.Name} - Level: [{e.Level}], EventId: [{e.EventId}], EventName: [{e.EventName}], Message: [{message}]");
```

Comment:
Would `WriteLine(String, Object[])` be better than string interpolation?

Reply:
In this case what do you mean by "better"? This is debug code only available in debug builds so I wasn't stressing the perf _at all_. I think whatever is most readable is probably "best" here 🤣 And this doesn't seem too bad for readability.

RE perf of string interpolation at least .NET 6 and great it isn't necessarily to be feared: https://devblogs.microsoft.com/dotnet/string-interpolation-in-c-10-and-net-6/. `Debug.WriteLine(string, object[])` may actually be worse due to the forced array allocation.

Reply:
I wasn't looking deep into this, this is where I came from "if Debug.WriteLine has multiple overloads, when would it be proper to use this versus that".

Reply:
> I wasn't looking deep into this, this is where I came from "if Debug.WriteLine has multiple overloads, when would it be proper to use this versus that".

Good question that we should probably have an answer to for guiding our decisions. I don't have a reference on hand, but I think I recall a fair amount of "it depends" regarding when string interpolation vs. string format is better.

Reply:
It used to be more of an "it depends".  It's no2 pretty much always "use interpolated strings and the best things will happen".

Writing:
```C#
Debug.WriteLine($"{e.EventSource.Name} - Level: [{e.Level}], EventId: [{e.EventId}], EventName: [{e.EventName}], Message: [{message}]");
```
will allocate a single string.  Writing:
```C#
Debug.WriteLine($"{0} - Level: [{1}], EventId: [{2}], EventName: [{3}], Message: [{4}]", e.EventSource.Name, e.Level, e.EventId, e.EventName, message);
```
will allocate an object[] for the arguments as well as boxing both `e.Level` and `e.EventId` in order to store those value types into the array.  It'll also result in calling string.Format, which will have to parse the format string on each call, whereas with the interpolated string, the C# compiler (assuming you're targeting .NET 6+) will do all the parsing once at compile time and will use an interpolated string handler that will avoid all of those extraneous costs. (If you're targeting pre-.NET 6, e.g. netstandard2.0, the interpolated string will just compile down to a string.Format, anyway, and it'll end up performing exactly the same as if you using the `object[]`-based overload.

Of course, this is all in an argument to a `Debug.WriteLine`, and so evaporates in non-DEBUG builds, anyway.

```C#
using BenchmarkDotNet.Attributes;
using BenchmarkDotNet.Running;
using System.Diagnostics.Tracing;

BenchmarkSwitcher.FromAssembly(typeof(Tests).Assembly).Run(args);

[MemoryDiagnoser(false)]
public class Tests
{
    private string _eventSourceName = "System.Threading.Tasks.TplEventSource";
    private EventLevel _eventLevel = EventLevel.Informational;
    private int _eventId = 8;
    private string _eventName = "TaskStarted";
    private string _message = "Task {2} executing.";

    [Benchmark(Baseline = true)]
    public string StringFormat() => string.Format("{0} - Level: [{1}], EventId: [{2}], EventName: [{3}], Message: [{4}]", _eventSourceName, _eventLevel, _eventId, _eventName, _message);

    [Benchmark]
    public string Interpolated() => $"{_eventSourceName} - Level: [{_eventLevel}], EventId: [{_eventId}], EventName: [{_eventName}], Message: [{_message}]";
}
```

|       Method |      Mean |    Error |   StdDev | Ratio | Allocated | Alloc Ratio |
|------------- |----------:|---------:|---------:|------:|----------:|------------:|
| StringFormat | 189.79 ns | 1.260 ns | 1.179 ns |  1.00 |     408 B |        1.00 |
| Interpolated |  84.43 ns | 0.878 ns | 0.778 ns |  0.44 |     296 B |        0.73 |